src/utilities/collection-utils.ts

```diff
@@ -51,15 +51,15 @@ const _equalsBy = function<T, V>(
  * Uses ... syntax to allow a single collection or multiple collections to be passed in, ie
  * CollectionUtils.hasValues([]) or CollectionUtils.hasValues([], [], [])
  *
- * @param {(...Array<(any[] | List<any>)} collections
+ * @param {(...Array<(any[] | Immutable.List<any>)} collections
  * @returns {boolean} False if `collections` is null/undefined, or every element is also null/undefined,
  * or has no sub-elements. True if any element has sub-elements.
  */
 const _hasValues = (
-    ...collections: Array<any[] | List<any> | undefined>
+    ...collections: Array<any[] | Immutable.List<any> | undefined>
```

Comment:
all over this file, why the `Immutable.` prefix? I'll stop spamming though.

Reply:
See previous comment